influxdb3_py_api/src/lib.rs

```diff
@@ -0,0 +1,166 @@
+    rows: Vec<Row>,
+    current_index: usize,
+}
+
+#[pymethods]
+impl PyWriteBatchIterator {
+    fn next_point(&mut self) -> PyResult<Option<PyObject>> {
+        if self.current_index >= self.rows.len() {
+            return Ok(None);
+        }
+
+        Python::with_gil(|py| {
+            let row = &self.rows[self.current_index];
+            self.current_index += 1;
+
+            // Import Point class
+            let point_class = py
+                .import("influxdb_client_3.write_client.client.write.point")?
+                .getattr("Point")
+                .unwrap();
+
+            // Create new Point instance with measurement name (table name)
+            let point = point_class.call1((self.table_definition.table_name.as_ref(),))?;
+
+            // Set timestamp
+            point.call_method1("time", (row.time,))?;
+
+            // Add fields based on column definitions and field data
+            for field in &row.fields {
+                if let Some(col_def) = self.table_definition.columns.get(&field.id) {
+                    let field_name = col_def.name.as_ref();
+
+                    match col_def.data_type {
+                        InfluxColumnType::Tag => {
+                            let FieldData::Tag(tag) = &field.value else {
+                                // error out because we expect a tag
+                                return Err(PyValueError::new_err(format!(
+                                    "expect FieldData:Tag for tagged columns, not ${:?}",
+                                    field
+                                )));
+                            };
+                            point.call_method1("tag", (field_name, tag.as_str()))?;
+                        }
+                        InfluxColumnType::Timestamp => {}
+                        // Do we need to check that the types match? Should already be validated.
+                        InfluxColumnType::Field(_) => {
+                            match &field.value {
+                                FieldData::String(s) => {
+                                    point.call_method1("field", (field_name, s.as_str()))?
+                                }
+                                FieldData::Integer(i) => {
+                                    point.call_method1("field", (field_name, *i))?
+                                }
+                                FieldData::UInteger(u) => {
+                                    point.call_method1("field", (field_name, *u))?
+                                }
+                                FieldData::Float(f) => {
+                                    point.call_method1("field", (field_name, *f))?
+                                }
+                                FieldData::Boolean(b) => {
+                                    point.call_method1("field", (field_name, *b))?
+                                }
+                                // TODO: Check what these should be
```

Comment:
tags are just strings, at least to the python API. They come in LP as strings, and get dictionary encoded in Arrow and Parquet.